src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/utilities/HeapHprofBinWriter.java

```diff
@@ -447,10 +470,11 @@ public synchronized void write(String fileName) throws IOException {
         // flush buffer stream.
         out.flush();
 
-        // Fill in final length
-        fillInHeapRecordLength();
-
-        if (useSegmentedHeapDump) {
+        if (!useSegmentedHeapDump) {
+            // Fill in final length
+            fillInHeapRecordLength();
```

Comment:
It's unclear to me why this code is now conditional on not using a segmented heap dump.

Reply:
Hi Chris,
Thanks for review.  The original implementation of heap dump (when not using segments) is to treat all HPROF_HEAP_DUMP section as a whole segment and fill the size after the data are ready to be flush. 
As specified at link: http://hg.openjdk.java.net/jdk6/jdk6/jdk/raw-file/tip/src/share/demo/jvmti/hprof/manual.html#mozTocId848088 .
so here if it is not segmented heap dump, it still need call fillInHeapRecordLength() to fill the size slot before data flush.

Reply:
ok